Add 2022-06-16 agenda and minutes (#61)
solid/notifications-panel: meetings/2022-06-16.md

#### dokieli
* SC: [dokieli](https://github.com/linkeddata/dokieli) looks for `solid:storageDescription` (experimenting with this discovery until https://github.com/solid/notifications/issues/58 is resolved) to discover notification channel/subscription information).

> * SC: Implemented a view for "Storage details" - not particularly interesting/useful to the user in the context of notification stuff, but it was to show that the application is aware of what's available at this point.

Reviewer: ```suggestion
* SC: Implemented a view for "Storage details" — not particularly interesting/useful to the user in the context of notification stuff, but it was to show that the application is aware of what's available at this point.
```

* SC: Simple screencast at https://github.com/solid/specification/issues/355#issuecomment-1140508784 along with other findings in the comment.

> * SC: Also implemented subscription request to `WebSocketSubscription2021` and connect to `notify:source`. I didn't yet implement an actual feature to use this - will share that in the future. Need server implementations to move forward in parallel to get anywhere with this. (Also need to have dokieli use Solid OIDC but this is a side issue.)

Reviewer: ```suggestion
* SC: Also implemented subscription request to `WebSocketSubscription2021` and connect to `notify:source`. I didn't yet implement an actual feature to use this — will share that in the future. Need server implementations to move forward in parallel to get anywhere with this. (Also need to have dokieli use Solid OIDC but this is a side issue.)
```

* SC: Left comments/issues/PRs based on this.
* eP: If GET or HEAD includes that Link relation on every resource that responds with 2xx response that seems to solve the problem.
* SC: May not be limited to 2xx.

### Discovery of Notification Subscription Metadata
URL: https://github.com/solid/notifications/issues/58

* SC: I'll summarise https://github.com/solid/notifications/issues/58#issuecomment-1151117524 .

> * CB: Each resource points to a notification discovery resource or if every resource in a container is managed by the same channel. If I subscribe to a container, I get a notification about the container. Is that intended to be the same or buggy?

Reviewer: First sentence doesn't parse...

Author: @uvdsl , would you like to clarify this?

* CB: question was asked if every resource in the storage points to the same description resource with information about subscribing to notification channels

> * SC: Generally when the resource description changes. The assumption is that you get notification, so if the containement information changes you would get a notification.

Reviewer: ```suggestion
* SC: Generally, the assumption is that you get a notification when the resource description changes, so if the containment information changes you would get a notification.
```

* SC: Forget about the prior/old/insecure websocket behaviour that's not the one described here.
* eP: We may need to document use cases to subscribe to contained resources

> * RG: How much of this should be on the server and how much on the client? Have we thought about how to do this in terms of implementation efficiency and balancing that against with the ease of making a client.

Reviewer: ```suggestion
* RG: How much of this should be on the server, and how much on the client? Have we thought about how to do this in terms of implementation efficiency, and balancing that against/with the ease of making a client?
```

> * SC: Not that I know of anything that's documented. Which is why we are hoping for implementation feedback to approach those questions.
> * ...: Myself I understood the protocol better after implementing some of it, and what things needs to be clarified in the spec or missing.

Reviewer: ```suggestion
* SC: Not that I know of, anything that's documented...which is why we are hoping for implementation feedback to approach those questions.
* ...: Myself, I understood the protocol better after implementing some of it, and what things needs to be clarified or are missing in the spec.
```

> * CB: Rahul has a point, the client side should always work, however if you imagine a web app that accesses a lot of resources on the server, it may affect scalability to have separate websockets connections.

Reviewer: ```suggestion
* CB: Rahul has a point, the client side should always work. However, if you imagine a web app that accesses a lot of resources on the server, it may affect scalability to have separate websocket connections.
```

> * eP: We should start with UC and track. With websocket connection, each resource requries subscription, the server could use source and just response - that's why we are optimising it. For the second one is an option for server to provide or not for aggregate subscription. It is mostly the container, as opposed to multiple topic.

Reviewer: ```suggestion
* eP: We should start with UC and track. With websocket connection, each resource requires subscription. The server could use source and just respond — that's why we are optimising it. For the second one, it is an option for server to provide aggregate subscription or not. It is mostly the container, as opposed to multiple topic.
```

* SC: My understanding was that as soon as the description of the resource changes, it can trigger a notification.

> * eP: Chat room container, a new chat. But edited a chat line, I get a notification just by subscribing to a container. In a way a propogation.

Reviewer: ```suggestion
* eP: Chat room container, a new chat. But editing a chat line, I get a notification just by subscribing to a container. In a way, a propagation.
```

> * SC: I think we do need to cicle back on that a bit with the Solid Protocol because there is this bit https://solidproject.org/TR/protocol#contained-resource-metadata . Lets come back to this.
> * SC: Right now Notification Protocol doesn't have a notion of Solid Storage. If we don't have that requirements where the channel descriptions are available from.

Reviewer: ```suggestion
* SC: I think we do need to cicle back on that a bit with the Solid Protocol because there is this bit: <https://solidproject.org/TR/protocol#contained-resource-metadata>. Lets come back to this.
* SC: Right now, Notification Protocol doesn't have a notion of Solid Storage. If we don't have that requirement, where are the channel descriptions available from?
```

* eP: The spec is called *Solid Notifications Protocol*, I think we can depend on Solid Protocol here.

> * SC: The discussion is if we define discovery in Notifications Protocol or we lean on Solid Protocol. * * ...: Notion of storage is the only thing that appears to be the dependency based on the current WIP.

Reviewer: ```suggestion
* SC: The discussion is whether we define discovery in Notifications Protocol or we lean on Solid Protocol. 
  * ...: Notion of storage is the only thing that appears to be the dependency, based on the current WIP.
```

> * RG: I think some notion of containment will be needed. We want to extend solid notifications like solid/solid-rest so that we can use solid with say, the filesystem. 

Reviewer: ```suggestion
* RG: I think some notion of containment will be needed. We want to extend solid notifications like solid/solid-rest so that we can use solid with, say, the filesystem. 
```

> * RG: Could you clarify which part of the solid protocols do you want to lean on to inform the solid-notifications vis-a-vis the containment issue.

Reviewer: ```suggestion
* RG: Could you clarify which part of the solid protocols you want to lean on to inform the solid-notifications vis-a-vis the containment issue?
```

> * SC: I don't think there is expectation that you get notification that resource has changed but not neccessary on what has changed. There is also authorization aspec, one may have access to description of a container but not all contained resources.

Reviewer: ```suggestion
* SC: I think there is expectation that you get a notification that a resource has changed, but not necessarily on what has changed. There is also authorization aspect — one may have access to the description of a container, but not to all contained resources.
```

* RG: Even if you don't have access to a contained document, would you get notification?
* SC: Yes but not what changed within the contained resource. No different than doing a GET/HEAD directly on the container - whatever is observed there.

### Next release of Notifications Protocol

> * SC: Solid Protocol 0.9.1 will be released with https://github.com/specification/issue/355 resolved and will require (MUST) Notifications Protocol (NP). So, we should resolve https://github.com/solid/notifications/issues/58 (along with some other low hanging stuff and other dependencies) and publish NP-next prior to Solid Protocol 0.9.1.

Reviewer: ```suggestion
* SC: Solid Protocol 0.9.1 will be released with https://github.com/specification/issue/355 resolved and will require (`MUST`) Notifications Protocol (NP). So, we should resolve https://github.com/solid/notifications/issues/58 (along with some other low hanging fruit and other dependencies), and publish NP-next prior to Solid Protocol 0.9.1.
```
